### ISSUE TITLE
PlatformException (PlatformException(0, Unknown error,  After Payment Processing ... 

### ISSUE THREAD

**Issue by sagarshende23** [paytm/paytm-allinonesdk-flutter#15]
**Getting this Exception Unkown Error** 

`PlatformException (PlatformException(0, Unknown error, {response: {"BANKNAME":"State Bank of India","BANKTXNID":"14835104078","CHECKSUMHASH":"cGPIPF6jwg2jhv8LnaxpZqQe5ipfjXMt3Ce20KoLAiKtAouMPaD4YWXrD+HoVlLzLCsfUckpN3FUkpsehg4+KbFpz9q3zao7BgVJgeXAMng\u003d","CURRENCY":"INR","GATEWAYNAME":"SBI","MID":"IqXcPG22000933596044","ORDERID":"1ab36f43-8397-4dc1-9b4f-7fca41f22858","PAYMENTMODE":"NB","RESPCODE":"01","RESPMSG":"Txn Success","STATUS":"TXN_SUCCESS","TXNAMOUNT":"210.00","TXNDATE":"2021-09-11 11:17:48.0","TXNID":"20210911111212800110168378902983217"}}, null))`

**But Transaction is Successful ** 

`"STATUS":"TXN_SUCCESS"`

Is any Problem With CallbackUrl...

Why App is Getting an Exception??

**Comment by soumyavats** [comment]
Hi Can you please share the callback URL which you are using?

**Comment by sagarshende23** [comment]
Sorry for Late Response, Call Back Url is, I'm using an custom backend url...to get Data as Response in backend

**Comment by soumyavats** [comment]
Please use our static callback url and check once

**Comment by sagarshende23** [comment]
YES, it's working but how can we Include our own Callback url...to get the response?

**Comment by soumyavats** [comment]
it seems there is some issue with your callback url

**Comment by sagarshende23** [comment]
If someone wants to use their own callback url for getting response from paytm, how can they set it up ? Or is that not possible as of now on Paytm allinoneSDK?

**Comment by soumyavats** [comment]
Dynamic callback url is also supported, but it seems there is some issue with your callback url, what functionalities you are doing on the callback URL?

**Comment by rahulbansal7** [comment]
Hi @soumyavats, I am working with @sagarshende23. The callback url triggers an API at our server, verifies order id and then sends back response. I can see that you tried to trigger the callback url that threw you an error (possibly 500).
Flow of our callback url => 

- Receive request, process body as form data. 
- Trigger API on our server and validate order id
- Send back html form that is same as your static callback url

We need your help to set it up as in tell us what response we should send from our dynamic callback url so that in flutter it works the same as your static url.

**Comment by soumyavats** [comment]
Hi @rahulbansal7 you can Trigger API on your server and call the validate order id once you receive the callback in the promise on frontend, that would work for you.  This is the recommended way of integrating

**Comment by rahulbansal7** [comment]
@soumyavats Got it. Thanks. So we have to use the static URL in any case. But how can I get a server to server response so I can update our db? I have set up the webhook url as mentioned in the docs but that webhook url (same as our callback url) never gets triggered. Is there any way to get a s2s response for payment?

**Comment by soumyavats** [comment]
you can raise a query on the Paytm developer support section if your webhook is not working. And for updating, your db you can call your API which is validating the order, and there you can call transaction status API

**Comment by rahulbansal7** [comment]
@soumyavats  Thanks for the help! One more question, In the faqs section (https://developer.paytm.com/docs/callback-and-webhookurl-faq/?ref=callbackWebhook)  its says -> 
`No these are different. Callback URL is the one which helps you get the transaction response (Success/Fail/Pending) after completing the transaction. Webhook - For cases where a transaction was originally reported as Pending due to some reason at paytm/bank end and subsequently gets successful or failed. Paytm will post the final transaction response back to the merchant through a webhook.`

As per my understanding does this mean we only get a s2s response in case of pending transactions? or does the webhook get triggered as soon as the payment is made?

**Comment by soumyavats** [comment]
In every case you will get s2s response on webhook

**Comment by rahulbansal7** [comment]
So the problem seems to be with the webhook setup. @soumyavats Thanks for the support!

**Comment by sagarshende23** [comment]
HI @soumyavats, Why I'm Getting an Error for Setting  `bool restrictAppInvoke = false;`


`Error PlatformException (PlatformException(0, App Invoke is not allowed for this merchant, null, null))
`


![Screenshot 2021-09-09 at 9 54 08 PM](https://user-images.githubusercontent.com/43273993/134629465-cfe33098-0b3f-4708-a3b8-7ba30dc79efa.jpeg)

Note:- **Paytm App is Installed in the Phone (Android device)**

**Comment by soumyavats** [comment]
You are testing this on staging or on production, if it's staging mid please install Paytm staging app from github

**Comment by sagarshende23** [comment]
Thanks For Help... @soumyavats
